src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java

```diff
@@ -379,29 +368,53 @@ private RequestOptions getRequestOptions(PartitionKey key, Integer requestUnit)
 
     private <T> List<T> executeQuery(@NonNull com.microsoft.azure.documentdb.SqlQuerySpec sqlQuerySpec,
                                      boolean isCrossPartition, @NonNull Class<T> domainClass, String collectionName) {
-        final FeedResponse<Document> feedResponse = executeQuery(sqlQuerySpec, isCrossPartition, collectionName);
+        final com.microsoft.azure.documentdb.FeedResponse<Document> feedResponse =
+                executeQuery(sqlQuerySpec, isCrossPartition, collectionName);
         final List<Document> result = feedResponse.getQueryIterable().toList();
 
         return result.stream().map(r -> getConverter().read(domainClass, r)).collect(Collectors.toList());
     }
 
-    private FeedResponse<Document> executeQuery(@NonNull com.microsoft.azure.documentdb.SqlQuerySpec sqlQuerySpec,
-                                                boolean isCrossPartition, String collectionName) {
-        final FeedOptions feedOptions = new FeedOptions();
-        final String selfLink = toCollectionSelfLink(collectionName);
+    private com.microsoft.azure.documentdb.FeedResponse<Document> executeQuery(
+            @NonNull com.microsoft.azure.documentdb.SqlQuerySpec sqlQuerySpec,
+            boolean isCrossPartition, String collectionName) {
+        final com.microsoft.azure.documentdb.FeedOptions feedOptions = new com.microsoft.azure.documentdb.FeedOptions();
+        final String selfLink = getCollectionLink(collectionName);
 
         feedOptions.setEnableCrossPartitionQuery(isCrossPartition);
 
         return getDocumentClient().queryDocuments(selfLink, sqlQuerySpec, feedOptions);
     }
 
-    private FeedResponse<Document> executeQuery(@NonNull com.microsoft.azure.documentdb.SqlQuerySpec sqlQuerySpec,
-                                                FeedOptions feedOptions, String collectionName) {
-        final String selfLink = toCollectionSelfLink(collectionName);
+    private com.microsoft.azure.documentdb.FeedResponse<Document> executeQuery(
+            @NonNull com.microsoft.azure.documentdb.SqlQuerySpec sqlQuerySpec,
+            com.microsoft.azure.documentdb.FeedOptions feedOptions,
+            String collectionName) {
+        final String selfLink = getCollectionLink(collectionName);
 
         return getDocumentClient().queryDocuments(selfLink, sqlQuerySpec, feedOptions);
     }
 
+    // Use public for now, will change to private after referenced
+    public Observable<com.microsoft.azure.cosmosdb.Document> executeQueryAsyncDocument(
+            @NonNull SqlQuerySpec sqlQuerySpec, @NonNull String collectionName, @NonNull FeedOptions options) {
+        return executeQueryAsync(sqlQuerySpec, collectionName, options)
+                .subscribeOn(Schedulers.io())
+                .map(FeedResponse::getResults)
+                .subscribeOn(Schedulers.io())
+                .flatMap(Observable::from);
+    }
+
+    // Use public for now, will change to private after referenced
```

Comment:
Only add two basic executeQuery method for Async. we can add more if need base on this.